packages/task/src/browser/task-terminal-widget-manager.ts

```diff
@@ -0,0 +1,184 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { ApplicationShell, WidgetOpenerOptions } from '@theia/core/lib/browser';
+import { TerminalWidget } from '@theia/terminal/lib/browser/base/terminal-widget';
+import { TerminalWidgetFactoryOptions } from '@theia/terminal/lib/browser/terminal-widget-impl';
+import { TerminalService } from '@theia/terminal/lib/browser/base/terminal-service';
+import { PanelKind, TaskConfiguration, TaskWatcher, TaskExitedEvent, TaskServer } from '../common';
+import { TaskDefinitionRegistry } from './task-definition-registry';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+
+export interface TaskTerminalWidgetOpenerOptions extends WidgetOpenerOptions, TerminalWidgetFactoryOptions {
+    taskId: number;
+    taskConfig?: TaskConfiguration;
+}
+
+export interface TaskTerminalWidget extends TerminalWidget {
+    readonly kind: 'task';
+    dedicated?: boolean;
+    taskId?: number;
+    taskConfig?: TaskConfiguration;
+    busy?: boolean;
```

Comment:
It would  be good to add js-doc of what you mean by busy looking at the PR we seem to have different understanding.

For me busy is whether associated task is running, it should be set when a task starts running and be unset when a tasks stop. It should not be changed by other events.
PR though sets it on `didConnect`, I don't understand why it is necessary and relevant.

Reply:
I see why `busy` means same way as you do.
Please see the reason of using `onTerminalDidConnectProcess` from https://github.com/eclipse-theia/theia/pull/7260#discussion_r396827368